src/csvcubed/inspect/browsercomponents.py

```diff
@@ -0,0 +1,112 @@
+from dataclasses import dataclass, field
+
+from csvcubed.inspect.lazyfuncdescriptor import lazy_func_field
+from csvcubed.models.sparqlresults import QubeComponentResult
+
+
+@dataclass(frozen=True)
+class Dimension:
+    dimension_component: QubeComponentResult = field(repr=False)
+
+    def _get_dimension_uri(self) -> str:
+        return self.dimension_component.property
+
+    dimension_uri: str = lazy_func_field(_get_dimension_uri)
+
+
+@dataclass(frozen=True)
+class ExternalDimension(Dimension):
+    """A dimension defined externally to this data set."""
+
+    pass
+
+
+@dataclass(frozen=True)
+class LocalDimension(Dimension):
+    """A dimension which is defined locally in this data set."""
+
+    def _get_label(self):
+        dimension_label = self.dimension_component.property_label
+        if dimension_label is None:
+            raise ValueError("Could not locate label for locally defined dimension")
+
+        return dimension_label
+
+    label: str = lazy_func_field(_get_label)
+
+    # todo: Should bring parent/subPropertyOf value in here.
+
+
+@dataclass(frozen=True)
+class Attribute:
+    attribute_component: QubeComponentResult = field(repr=False)
+
+    def _get_attribute_uri(self) -> str:
+        return self.attribute_component.property
+
+    attribute_uri: str = lazy_func_field(_get_attribute_uri)
+
+
+@dataclass(frozen=True)
+class ExternalAttribute(Attribute):
```

Comment:
Should we have separate components for External/Local Attribute Resource/Literals?

We can determine the data type of the data in the column for both Internal/ExternalAttributeLiterals which may be helpful in understanding the data.